orc8r/gateway/python/scripts/register.py

```diff
@@ -0,0 +1,128 @@
+#!/usr/bin/env python3
+"""
+Copyright 2020 The Magma Authors.
+
+This source code is licensed under the BSD-style license found in the
+LICENSE file in the root directory of this source tree.
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import argparse
+import sys
+import textwrap
+from typing import List
+
+import checkin_cli
+import snowflake
+from magma.common.cert_utils import load_public_key_to_base64der
+from magma.common.service_registry import ServiceRegistry
+from magma.configuration.service_configs import save_override_config
+from orc8r.protos.bootstrapper_pb2 import (
+    ChallengeKey,
+    RegisterRequest,
+    RegisterResponse,
+)
+from orc8r.protos.bootstrapper_pb2_grpc import RegistrationStub
+from orc8r.protos.identity_pb2 import AccessGatewayID
+
+
+def register_handler(client: RegistrationStub, args: List[str]) -> RegisterResponse:
+    """
+    Register a device and retrieves its control proxy
+    Args:
+        client: Registration stub
+        args: command line arguments
+    Returns:
+        RegisterResponse: response from gRPC call, either error or the control_proxy
+    """
+    req = RegisterRequest(
+        token=args.token,
+        hwid=AccessGatewayID(
+            id=snowflake.snowflake(),
+        ),
+        challenge_key=ChallengeKey(
+            key=load_public_key_to_base64der("/var/opt/magma/certs/gw_challenge.key"),
+            key_type=ChallengeKey.KeyType.SOFTWARE_ECDSA_SHA256,
+        ),
+    )
+
+    res = client.Register(req)
+    if res.HasField("error"):
+        raise Exception(res.error)
+
+    msg = textwrap.dedent(
```

Comment:
Nit: I'd prob put this printed msg in the main func, to keep this func as "stateless" as possible